internal/services/managedapplications/managed_application_resource.go

```diff
@@ -394,21 +433,51 @@ func flattenManagedApplicationParameterValuesValueToString(input *interface{}) (
 		for k, v := range attrs.(map[string]interface{}) {
 			if v != nil {
 				delete(attrs.(map[string]interface{})[k].(map[string]interface{}), "type")
+
+				// Secure values are not returned, thus settings it with local value
+				value := attrs.(map[string]interface{})[k].(map[string]interface{})
+				if _, ok := value["value"]; !ok {
+					value["value"] = ""
+					if localParam, localParamOK := localParameters[k]; localParamOK {
+						if localParamValue, localParamValueOK := localParam.(map[string]interface{})["value"]; localParamValueOK {
```

Comment:
As above, we should add more type-checking here

Reply:
`localParameters` comes from expanded blocks from terraform sdk, not from api side so we shall be able to asure it is a map.

Reply:
yes make sense, have added the type check